core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java

```diff
@@ -228,19 +232,24 @@ public FlowIdentifier updateNodeCount(Stack stack, InstanceGroupAdjustmentV4Requ
         environmentService.checkEnvironmentStatus(stack, EnvironmentStatus.upscalable());
         try {
             return transactionService.required(() -> {
+                boolean upscale = instanceGroupAdjustmentJson.getScalingAdjustment() > 0;
                 Stack stackWithLists = stackService.getByIdWithLists(stack.getId());
                 updateNodeCountValidator.validateServiceRoles(stackWithLists, instanceGroupAdjustmentJson);
                 updateNodeCountValidator.validateStackStatus(stackWithLists);
                 updateNodeCountValidator.validateInstanceGroup(stackWithLists, instanceGroupAdjustmentJson.getInstanceGroup());
                 updateNodeCountValidator.validateScalabilityOfInstanceGroup(stackWithLists, instanceGroupAdjustmentJson);
                 updateNodeCountValidator.validateScalingAdjustment(instanceGroupAdjustmentJson, stackWithLists);
-                updateNodeCountValidator.validateInstanceStatuses(stackWithLists, instanceGroupAdjustmentJson);
+                if (!upscale || !targetedUpscaleSupportService.targetedUpscaleOperationSupported(stackWithLists)) {
```

Comment:
maybe it would worth storing the result of the condition in a variable as it is reused few lines later

Reply:
the result is cached but yes, I'll extract it